ui/ducks/send/send.js

```diff
@@ -2354,15 +2355,66 @@ export function updateSendAsset(
       }
     } else {
       await dispatch(showLoadingIndication());
-      const details = {
-        ...providedDetails,
-        ...(await getTokenStandardAndDetails(
-          providedDetails.address,
-          sendingAddress,
-          providedDetails.tokenId,
-        )),
+
+      const STANDARD_TO_REQUIRED_PROPERTIES = {
+        // 'balance' must be last so that we know all other properties exist if `missingProperty` = 'balance'
+        [TokenStandard.ERC20]: isReceived
+          ? ['address', 'symbol', 'decimals']
+          : ['address', 'symbol', 'decimals', 'balance'],
+        [TokenStandard.ERC721]: ['address', 'symbol', 'tokenId'],
+        [TokenStandard.ERC1155]: ['address', 'symbol', 'tokenId'],
       };
 
+      let missingProperty = STANDARD_TO_REQUIRED_PROPERTIES[
+        providedDetails.standard
+      ].find((property) => providedDetails[property] === undefined);
```

Comment:
Is the intent here to find all missing properties? Or do we know that there's always at most 1?

Reply:
It's to find if there are any, with special logic if it's only `balance`

Reply:
`find` only returns the first match so if an ERC20 token is also missing `decimals`, the existence of balance won't be checked. I think you need `findAll` here

Reply:
Yeah finding the first match is intended behavior. `getTokenStandardAndDetails` fetched all those missing properties (for context, before this change (f2e3e2ec1faaec5a7174c7cbabf40ac1598a1a67) we just called it all the time anyways. However, if only balance is needed – intentionally left at the end so we know that it's the only `missingProperty` if set – we can use a lighter API call.